### PR TITLE
chore!: Rename job config history plugin

### PR DESCRIPTION
**Pull request by NotMyFault**
# Description

Request to rename https://github.com/jenkinsci/job-config-history-plugin

# Submitter checklist for adding or changing permissions

### Always

- [X] Add link to plugin/component Git repository in description above

### For a newly hosted plugin only

- [ ] Add link to resolved HOSTING issue in description above

### For a new permissions file only

- [ ] Make sure the file is created in `permissions/` directory
- [ ] `artifactId` (pom.xml) is used for `name` (permissions YAML file).
- [ ] [`groupId` / `artifactId` (pom.xml) are correctly represented in `path` (permissions YAML file)](https://github.com/jenkins-infra/repository-permissions-updater/#managing-permissions)
- [ ] Check that the file is named `plugin-${artifactId}.yml` for plugins

### When adding new uploaders (this includes newly created permissions files)

- [ ] [Make sure to `@`mention an existing maintainer to confirm the permissions request, if applicable](https://github.com/jenkins-infra/repository-permissions-updater/#requesting-permissions)
- [ ] Use the Jenkins community (LDAP) account name in the YAML file, not the GitHub account name
- [ ] Make sure to `@`mention the users being added so their GitHub account names are known if they require GitHub merge access (see below).
- [ ] [All newly added users have logged in to Artifactory at least once](https://github.com/jenkins-infra/repository-permissions-updater/#requesting-permissions)

### Reviewer checklist (not for requesters!)

- [ ] Check this if newly added person also needs to be given merge permission to the GitHub repo (please @ the people/person with their GitHub username in this issue as well). If needed, it can be done using an [IRC Bot command](https://jenkins.io/projects/infrastructure/ircbot/#github-repo-management)
- [ ] Check that the `$pluginId Developers` team has `Admin` permissions while granting the access.
- [ ] In the case of plugin adoption, ensure that the Jenkins Jira default assignee is either removed or changed to the new maintainer.
- [ ] If security contacts are changed (this includes add/remove), ping the security officer (currently `@Wadeck`) in this pull request. If an email contact is changed, wait for approval from the security officer.

There are [IRC Bot commands](https://jenkins.io/projects/infrastructure/ircbot/#issue-tracker-management) for it

### CODE REVIEW

File: permissions/plugin-job-config-history.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+name: "job-config-history"
+github: "jenkinsci/job-config-history-plugin"
+issues:
+- jira: '15683' # jobconfighistory-plugin
+paths:
+- "org/jenkins-ci/plugins/job-config-history"
```
Comment (timja) [inline permissions/plugin-job-config-history.yml:7]: you can't rename the artifact ID as there's no way to upgrade from that but we can rename the repo that's fine, although you may want to keep the artifact id in the file title, i.e. not rename this file.
```suggestion
- "org/jenkins-ci/plugins/jobConfigHistory"
```

Comment (NotMyFault) [reply permissions/plugin-job-config-history.yml]: Good call, reverted that back 👍🏻

File: permissions/plugin-jobConfigHistory.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
-name: "jobConfigHistory"
-github: "jenkinsci/jobConfigHistory-plugin"
+name: "job-config-history"
```
Comment (timja) [inline permissions/plugin-jobConfigHistory.yml:2]: name == artifactId
https://github.com/jenkins-infra/repository-permissions-updater#managing-permissions

```suggestion
name: "jobConfigHistory"
```

Comment (NotMyFault) [reply permissions/plugin-jobConfigHistory.yml]: 🤦🏻 me

**Review by timja — APPROVED** [review]

**Comment by halkeye** [comment]
Make sure you update the <url> in your pom. The renamed repo means you won't have docs on plugins.jenkins.io until a new release is out with the url is fixed

**Comment by NotMyFault** [comment]
> Make sure you update the in your pom. The renamed repo means you won't have docs on plugins.jenkins.io until a new release is out with the url is fixed

Yeah, I'm heading forward to update the GitHub URL in a bit. Thanks for the reminder!

**Pull request merged**